### ISSUE TITLE
Numerical Problem Quantizing Model to 1 Bit

### ISSUE THREAD

**Issue by ylfzr** [Tencent/PocketFlow#73]
**Describe the bug**
Numerical problem occurs when quantize  ResNet56 model on Cifar10 dataset.

**To Reproduce**
1. Quantize  a pretrained ResNet56 model to 1 bit. 
2. Run the uniform/non-unifom learner, quantize and fintune, after a few iterations the accuracy drops to 0.1

**Expected behavior**
The accuracy goes up as fine-tuning proceeds.

**Desktop (please complete the following information):**
 - OS: Centos
 - Browser chrome
 - Version None

**screen shots **
![image](https://user-images.githubusercontent.com/22250974/48657416-3f590900-ea6b-11e8-8e60-bc049e8efdd6.png)

**Additional context**
The problem is caused by line 227 in learners/uniform_quantization/utils.py (for the non-uniform quantization the same problem arise) alpha = w_max - w_min. alpha for some weights might some how goes to zeros (or very near 0) during fine-tuning.  Causing an Nan problem. The problem can be solved by adding a small noise term. A PR has been sent.

**Comment by jiaxiang-wu** [comment]
Thanks for your contribution. The PR has been merged.
